indexing-service/src/main/java/org/apache/druid/indexing/common/actions/CheckPointDataSourceMetadataAction.java

```diff
@@ -68,13 +69,32 @@ public Integer getTaskGroupId()
   }
 
   // For backwards compatibility
-  @Deprecated
   @JsonProperty
-  public SeekableStreamDataSourceMetadata getPreviousCheckPoint()
+  private SeekableStreamDataSourceMetadata getPreviousCheckPoint()
   {
     return checkpointMetadata;
   }
 
+  // For backwards compatibility
+  @JsonProperty
+  private SeekableStreamDataSourceMetadata getCurrentCheckPoint()
+  {
+    return checkpointMetadata;
+  }
+
+  /**
+   * Returning a dummy value so the objects from older versions still work
```

Comment:
How would returning a dummy value make objects from older versions work with the current version? Modifying the serialization code now won't change what older versions do.

Reply:
This method is to add the missing property (`sequenceName`) in newer serialized JSON. So the task fails on older middlemanager since it requires this property, I think it doesn't matter what value is returned here, as long as the property is present since the value is not really used in any production code. But i guess the javadoc can be updated to say it's for rolling updates.